docs/topics/arrays.md

```diff
@@ -1,62 +1,447 @@
     val asc = Array(5) { i -> (i * i).toString() }
-    asc.forEach { println(it) }
+    asc.forEach { print(it) }
+    // 014916
 //sampleEnd
 }
 ```
-{kotlin-runnable="true" kotlin-min-compiler-version="1.3"}
+{kotlin-runnable="true" kotlin-min-compiler-version="1.3" id="arrays-array-constructor-kotlin"}
+
+> Indices start from 0 in Kotlin.
+>
+{type="note"}
+
+### Empty arrays
 
-The `[]` operation stands for calls to member functions `get()` and `set()`.
+To create an empty array, use an [`emptyArray()`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/empty-array.html):
 
-Arrays in Kotlin are _invariant_. This means that Kotlin does not let us assign an `Array<String>`
-to an `Array<Any>`, which prevents a possible runtime failure (but you can use `Array<out Any>`,
-see [Type Projections](generics.md#type-projections)).
+```kotlin
+    var exampleArray = emptyArray<String>()
+```
 
-## Primitive type arrays
+> You can specify the type of the empty array on the left-hand or right-hand side of the assignment due to Kotlin's type
+> inference.
+> 
+> For example:
+> ```Kotlin
+> var exampleArray = emptyArray<String>()
+> 
+> var exampleArray: Array<String> = emptyArray()
+>```
+>
+{type="note"}
 
-Kotlin also has classes that represent arrays of primitive types without boxing overhead: `ByteArray`,
-`ShortArray`, `IntArray`, and so on. These classes have no inheritance relation to the `Array` class, but they
-have the same set of methods and properties. Each of them also has a corresponding factory function:
+### Multidimensional arrays
+
+Arrays can be nested within each other to create multidimensional arrays:
 
 ```kotlin
-val x: IntArray = intArrayOf(1, 2, 3)
-x[0] = x[1] + x[2]
+fun main() {
+//sampleStart
+    // Creates a two-dimensional array
+    val twoDArray = Array(2) { Array<Int>(2) { 0 } }
+    println(twoDArray.contentDeepToString())
+    // [[0, 0], [0, 0]]
+
+    // Creates a three-dimensional array
+    val threeDArray = Array(3) { Array(3) { Array<Int>(3) { 0 } } }
+    println(threeDArray.contentDeepToString())
+    // [[[0, 0, 0], [0, 0, 0], [0, 0, 0]], [[0, 0, 0], [0, 0, 0], [0, 0, 0]], [[0, 0, 0], [0, 0, 0], [0, 0, 0]]]
+//sampleEnd
+}
+```
+{kotlin-runnable="true" kotlin-min-compiler-version="1.3" id="arrays-multidimensional-array-kotlin"}
+
+> Nested arrays don't have to be the same type or the same size.
+>
+{type="note"}
+
+## Access and modify elements
+
+Arrays are always mutable. To access and modify elements in an array, use the [indexed access operator](operator-overloading.md#indexed-access-operator)`[]`:
+
+```kotlin
+fun main() {
+//sampleStart
+    val simpleArray = arrayOf(1, 2, 3)
+    val twoDArray = Array(2) { Array<Int>(2) { 0 } }
+
+    // Accesses the element and modifies it
+    simpleArray[0] = 10
+    twoDArray[0][0] = 2
+
+    // Prints the modified element
+    println(simpleArray[0].toString()) // 10
+    println(twoDArray[0][0].toString()) // 2
+//sampleEnd
+}
```

Comment:
Since `[]` is translated to `.get()` and `.set()` functions, is there a benefit from covering these functions?

Reply:
I would not consider using `.get` and `.set` for arrays idiomatic Kotlin, so rather than give users the "choice" here, I'd nudge them towards using indexed access throughout the array docs.